app/src/main/java/net/squanchy/navigation/firststart/FirstStartWithNoNetworkActivity.java

```diff
@@ -0,0 +1,224 @@
+    }
+
+    private void swapProgressWithSuccessAndContinue() {
+        ctaProgressView.setImageResource(R.drawable.ic_circle_tick);
+
+        animate(ctaProgressView, popBackIn(), this::continueToScheduleAfterDelay)
+                .start();
+    }
+
+    private Func1<View, Animator> popBackIn() {
+        return view -> {
+            Animator alphaAnimator = createFadeAnimationFor(view, 0f, 1f);
+            Animator scaleAnimator = createScaleAnimationFor(
+                    view,
+                    new BounceInterpolator(),
+                    0f, 1f);
+            AnimatorSet animator = new AnimatorSet();
+            animator.playTogether(alphaAnimator, scaleAnimator);
+            animator.setDuration(shortAnimationDuration);
+            return animator;
+        };
+    }
+
+    private Animator createFadeAnimationFor(View view, float... values) {
+        Property<View, Float> property = Property.of(View.class, Float.class, "alpha");
+        ObjectAnimator animator = ObjectAnimator.ofFloat(view, property, values);
+        animator.setInterpolator(new LinearInterpolator());
+        return animator;
+    }
+
+    private Animator createScaleAnimationFor(View view, Interpolator interpolator, float... values) {
+        Property<View, Float> scaleX = Property.of(View.class, Float.class, "scaleX");
+        Property<View, Float> scaleY = Property.of(View.class, Float.class, "scaleY");
+        AnimatorSet animator = new AnimatorSet();
+        animator.playTogether(
+                ObjectAnimator.ofFloat(view, scaleX, values),
+                ObjectAnimator.ofFloat(view, scaleY, values)
+        );
+        animator.setInterpolator(interpolator);
+        return animator;
+    }
+
+    private Animator animate(View view, Func1<View, Animator> animationProducer, Action endAction) {
+        Animator animator = animationProducer.call(view);
+        animator.addListener(new AnimationEndListener() {
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                try {
+                    endAction.run();
+                } catch (Exception e) {
+                    Timber.e(e);
+                }
+            }
+        });
+        return animator;
+    }
+
+    private void continueToScheduleAfterDelay() {
+        ctaProgressView.postDelayed(() -> {
+            startActivity(continuationIntent);      // We don't use the navigator here, we basically want to restart the whole flow
+            finish();
+        }, DELAY_AFTER_ANIMATIONS_MILLIS);
+    }
+
+    private class NetworkConnectedCallback extends ConnectivityManager.NetworkCallback {
+
+        private boolean receivedOnAvailable = false;
+
+        @Override
+        public void onAvailable(Network network) {
+            if (receivedOnAvailable) {
+                return;
+            }
+
+            receivedOnAvailable = true;
+
+            handler.post(FirstStartWithNoNetworkActivity.this::showNetworkAcquiredAndFinish);
```

Comment:
Apparently, yes. Initially I didn't have the handler and the app crashed because of that.